tdrs-backend/tdpservice/users/models.py

```diff
@@ -42,6 +43,217 @@ class RegionMeta(models.Model):
+    """Model to track user information change requests."""
+
+    class Meta:
+        """Define meta attributes."""
+
+        ordering = ['-requested_at']
+        verbose_name = _('User Change Request')
+        verbose_name_plural = _('User Change Requests')
+
+    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
+    user = models.ForeignKey(
+        'users.User',
+        on_delete=models.CASCADE,
+        related_name='change_requests'
+    )
+    requested_by = models.ForeignKey(
+        'users.User',
+        on_delete=models.CASCADE,
+        related_name='requested_changes'
+    )
+    field_name = models.CharField(
+        max_length=100,
+        help_text=_('The name of the field being changed, possible values include: first_name, last_name, regions and has_fra_access')
+    )
+    current_value = models.TextField(
+        blank=True,
+        help_text=_('The current value of the field')
+    )
+    requested_value = models.TextField(
+        blank=True,
+        help_text=_('The requested new value for the field')
+    )
+    status = models.CharField(
+        max_length=20,
+        choices=UserChangeRequestStatus.choices,
+        default=UserChangeRequestStatus.PENDING,
+        help_text=_('The current status of this change request')
+    )
+    requested_at = models.DateTimeField(
+        auto_now_add=True,
+        help_text=_('When this change was requested')
+    )
+    reviewed_by = models.ForeignKey(
+        'users.User',
+        on_delete=models.SET_NULL,
+        null=True,
+        blank=True,
+        related_name='reviewed_changes',
+        help_text=_('The admin who reviewed this change request')
+    )
+    notes = models.TextField(
+        blank=True,
+        help_text=_('Admin notes on approval/rejection')
+    )
+
+    def __str__(self):
+        """Return string representation."""
+        return f"{self.user.username} - {self.field_name} - {self.get_status_display()}"
+
+    def approve(self, admin_user, notes=None):
+        """Approve the change request and apply changes to the user."""
+        if self.status != UserChangeRequestStatus.PENDING:
+            return False
+
+        user = self.user
+        field_name = self.field_name
+        new_value = self.requested_value
+
+        # Apply the change to the user
+        updated_fields = None
+        try:
+            if field_name == 'regions':
+                user.regions.remove(*user.regions.all())  # Clear existing regions
+                regions = ast.literal_eval(new_value)
+                for region in regions:
+                    user.regions.add(region)
+            elif field_name == 'has_fra_access':
+                fra_permission = Permission.objects.get(codename='has_fra_access')
+                if new_value == 'True':
+                    user.user_permissions.add(fra_permission)
+                else:
+                    user.user_permissions.remove(fra_permission)
+            else:
```

Comment:
could we split this function up? the more fields/permissions we add, the more complex it will become